BombsExcersice/BombsExcersice.cpp

```diff
@@ -1,10 +1,173 @@
-// BombsExcersice.cpp : This file contains the 'main' function. Program execution begins and ends there.
-//
-
 #include <iostream>
+#include <vector>
+#include <fstream>
+#include <string>
+
+enum Type {Empty, Bomb, Wall};
+
+class Cell
+{
+public:
+    Cell() : type{ Empty }, lifetime{ 0 }, maxLifetime{ 3 } {}
+
+    void setType(Type t) { type = t; }
+    Type getType() const { return type; }
+
+    void becomeEmpty()
+    {
+        type = Empty;
+        lifetime = 0;
+    }
+    
+    int getLifetime() const { return lifetime; }
+    int getMaxLifetime() const { return maxLifetime; }
+    void increaseLifetime() { ++lifetime; }
+
+private:
+    Type type;
+    int lifetime;
+    int maxLifetime;
+};
+
+class BombsField
+{
+public:
+    BombsField(const int& rows, const int& cols) : rows(rows), cols(cols), field(rows, std::vector<Cell>(cols)) {}
+
+    void print() const
+    {
+        std::ofstream outputFile("output.txt");
```

Comment:
This function would be more generic if you made the stream an argument of the function, like this: `void print(std::basic_ostream & ostream) const`. Using `basic_ostream` instead of `ofstream` allows you to easily substitute streams and switch between the filestream and the standard stream if needed.